### PR TITLE
Fix Slot

### PR DESCRIPTION
**Pull request by TZL0**
'2022-02-29 10:00', '2022-01-01 24:00' no longer accepted

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -112,8 +112,6 @@ slot, and doctor.
 
 * Slot must be in the format `yyyy-MM-dd HH:mm`, eg. `2022-11-12 13:00`.
 
-* If slot is entered with time `24:00`, it will be saved as the `00:00` of the next date.
```
Comment (guokweijie) [inline docs/UserGuide.md:115]: why is this removed?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -159,11 +157,13 @@ parameters already exists in HealthContact.
...
 
-* Amount must be positive number with at most 2 decimal places.
+* An amount must be positive number with at most 2 decimal places.
 
 * One appointment can be attached to no more than one bill.
 
-* Bill date must be in the format `yyyy-MM-dd`, eg. `2022-11-12`.
+* A bill date must be in the format `yyyy-MM-dd`, eg. `2022-11-12`.
+
+* A bill date before appointment date is allowed as there may be reservation fee.
```
Comment (guokweijie) [inline docs/UserGuide.md:166]: i think we can just write that it is user's responsibility to check if there are date or time clashes for patients and doctors

File: docs/UserGuide.md [hunk 3]
```diff
@@ -159,11 +157,13 @@ parameters already exists in HealthContact.
 
 Adds a bill attached to an appointment with input information including amount and bill date.
 
-* Amount must be positive number with at most 2 decimal places.
+* An amount must be positive number with at most 2 decimal places.
 
 * One appointment can be attached to no more than one bill.
 
-* Bill date must be in the format `yyyy-MM-dd`, eg. `2022-11-12`.
+* A bill date must be in the format `yyyy-MM-dd`, eg. `2022-11-12`.
```
Comment (guokweijie) [inline docs/UserGuide.md:164]: change bill date file also to check for valid date

**Review by guokweijie — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 1 above)
Comment (TZL0) [reply docs/UserGuide.md:115]: 24:00 no longer valid

File: src/main/java/seedu/address/model/appointment/Slot.java [hunk 4]
```diff
@@ -27,24 +28,24 @@ public class Slot {
     public Slot(String dateTime) {
         requireNonNull(dateTime);
         checkArgument(isValidDateTime(dateTime), MESSAGE_CONSTRAINTS);
-        this.localDateTime = LocalDateTime.parse(dateTime, DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm"));;
+        this.localDateTime = LocalDateTime.parse(dateTime, FORMATTER);
```
Comment (SpecOps2016) [inline src/main/java/seedu/address/model/appointment/Slot.java:31]: Code is cleaner now that FORMATTER variable has been decalred

File: src/main/java/seedu/address/model/appointment/Slot.java [hunk 5]
```diff
@@ -27,24 +28,24 @@ public class Slot {
...
     }
 
     /**
      * Returns true if a given string is a valid date time.
      */
-    public static boolean isValidDateTime(String test) {
+    public static boolean isValidDateTime(String input) {
```
Comment (SpecOps2016) [inline src/main/java/seedu/address/model/appointment/Slot.java:37]: Naming of variable has been changed to make more sense

File: src/test/java/seedu/address/model/appointment/SlotTest.java [hunk 6]
```diff
@@ -48,12 +48,24 @@ public void constructor_invalidSlot5_throwsIllegalArgumentException() {
         assertThrows(IllegalArgumentException.class, () -> new Slot(invalidSlot));
     }
 
+    @Test
```
Comment (SpecOps2016) [inline src/test/java/seedu/address/model/appointment/SlotTest.java:51]: Tests added seems legitimate

**Review by SpecOps2016 — APPROVED** [review]
LGTM

**Review by guokweijie — APPROVED** [review]
